### PR TITLE
feat: splash screen and App logo implemented

### LINKED ISSUE #57: Null check on Null value for profile screen
Faced problem in displaying profile screen despite correct logic being implemented in the tab view screen

### LINKED ISSUE #58: Suggestion: Loading Animation for Resonate
A loading animation for Resonate replacing the default flutter loading animation (splash screen  which provides a loading animation when Resonate starts currently a default flutter loading animation is being shown would customize it for Resonate )

Would like to work on it

### LINKED ISSUE #61: Add a splash screen
Currently when the app restarts, we get to see the login screen for a second or so even if we are logged in. This leads to bad user experience. A better approach is to have a splash screen until the authentication controller checks if the user is already logged in and then navigate accordingly.

### LINKED ISSUE #65: Need for Logo Icon for Resonate
An app logo icon I believe is needed for Resonate along with its design needs to be finalized I believe it would differ from the AOISSIE-Org logo.

Screenshot 2023-06-21 at 9 49 06 AM

Currently being the default flutter logo, after finalizing the logo the splash screen I believe might also have to be updated the finalized logo being at the center with the organizations logo being at bottom center

### PR DESCRIPTION
**Pull request by Aarush-Acharya**
closes #61 and fixes #57, fixed #58, resolves #65

Resolved display of login screen for a split second while app restarts instead a splash screen is shown 

Navigated to tab view screen after onboarding 

Resolved null check on a null value in the profile screen

### CODE REVIEW

**Review by chandansgowda — CHANGES_REQUESTED** [review]
@Aarush-Acharya The app crashes if you logout of the account and restart the app. This is possibly because AuthenticationController getting exposed once the splash screen closes. Please check and make necessary fix.

**Comment by Aarush-Acharya** [comment]
Understood on it

**Comment by Aarush-Acharya** [comment]
@chandansgowda The fix has been integrated seems that the logic to navigate the user to the login screen after the splash screen if the user hasn't logged in was not written, it was a fault from my side apologies

**Review by chandansgowda — APPROVED** [review]

**Comment by chandansgowda** [comment]
@Aarush-Acharya Looks good. Please share the link for the PR on our discord server also.

**Comment by Aarush-Acharya** [comment]
@chandansgowda understood and thank you for reviewing the commits, the splash screen still needs the changes discussed

**Comment by chandansgowda** [comment]
Okay then its better to get the redesign with all app icon assets changed.

**Comment by Aarush-Acharya** [comment]
Understood doing in the same PR

**Comment by Aarush-Acharya** [comment]
Hey @chandansgowda  app logo has been implemented moving on to redesign of splash screen

**Comment by Aarush-Acharya** [comment]
Hey @chandansgowda  app logo has been implemented moving on to redesign of splash screen

**Comment by Aarush-Acharya** [comment]
Hey @chandansgowda apologies for not confirming the redesigned UI screen prior to committing 

here it is 
https://github.com/AOSSIE-Org/Resonate/assets/92685647/b7f4f1a3-5537-4c2e-a7b7-3448efa99722

do let me know of any changes required

**Comment by chandansgowda** [comment]
@Aarush-Acharya Can we increase the size of resonate logo ? I feel the splash screen still looks plain. And reduce AOSSIE logo size.

Or another option is to have apssie and resonates logo in the center as a row with a divider. Check which one looks good.

**Comment by Aarush-Acharya** [comment]
@chandansgowda  thank you would do and revert asap

**Comment by Aarush-Acharya** [comment]
Hey @chandansgowda  apologies for the delay

here are the two screens implemented do let me know which one to continue with 

A. Divider
https://github.com/AOSSIE-Org/Resonate/assets/92685647/405cf949-9660-4667-b836-b0162ff9ae15


B.  Refactored Original 
https://github.com/AOSSIE-Org/Resonate/assets/92685647/14e4dd57-2bc4-4e19-af66-a4581e1cc566

**Comment by chandansgowda** [comment]
Divider version looks better.

**Comment by Aarush-Acharya** [comment]
Understood would commit that one

**Comment by chandansgowda** [comment]
@Aarush-Acharya Okayy

**Comment by Aarush-Acharya** [comment]
Hey @chandansgowda I believe the work for this PR is done do let me know if anything needs to be updated would try to find some new issue to contribute to form tomorrow onwards

File: lib/main.dart [hunk 1]
```diff
@@ -55,15 +50,14 @@ class MyApp extends StatelessWidget {
...
               side: BorderSide(width: 1.0, color: Colors.grey[800]!)
             ),
-            minimumSize: const Size.fromHeight(45), // NEW
+            minimumSize: const Size.fromHeight(45), 
           ),
         )
       ),
-      home: const LoginScreen(),
-      initialBinding: AuthenticationBinding(),
+      initialRoute: '/splash',
```
Comment (chandansgowda) [inline lib/main.dart:60]: Use static routename from Approute class.

File: lib/bindings/auth_splash_bindings.dart [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 import 'package:resonate/controllers/authentication_controller.dart';
+import 'package:resonate/controllers/splash_controller.dart';
 
-class AuthenticationBinding extends Bindings{
+class AuthSplashBinding extends Bindings {
   @override
-  void dependencies(){
+  void dependencies() {
     Get.lazyPut(() => AuthenticationController());
+    Get.put(SplashController());
```
Comment (chandansgowda) [inline lib/bindings/auth_splash_bindings.dart:9]: Do we actually need to inject the controller compuslorily (put)? If yes, please explain why can't we use lazyPut.

**Review by chandansgowda — CHANGES_REQUESTED** [review]

File: lib/main.dart (hunk 1 above)
Comment (Aarush-Acharya) [reply lib/main.dart:60]: Understood

File: lib/bindings/auth_splash_bindings.dart (hunk 2 above)
Comment (Aarush-Acharya) [reply lib/bindings/auth_splash_bindings.dart:9]: While debugging tried to switch between put and lazy put but that wasn't the problem ig, still would test with lazy put and if it works as required would use lazy put then

Comment (Aarush-Acharya) [reply lib/bindings/auth_splash_bindings.dart:9]: Tried it out works fine

**Comment by Aarush-Acharya** [comment]
@chandansgowda the changes have been implemented

**Review by chandansgowda — APPROVED** [review]

**Pull request closed without merging**